runtime/altair/src/lib.rs

```diff
@@ -1666,8 +1666,32 @@ impl_runtime_apis! {
 			let pool = pallet_pool_system::Pool::<Runtime>::get(pool_id)?;
 			pool.tranches.tranche_currency(tranche_loc).map(Into::into)
 		}
+
+		fn portfolio_valuation(pool_id: PoolId) -> Option<Balance> {
+			if !pallet_pool_system::Pool::<Runtime>::get(pool_id).is_some() {
```

Comment:
No completely sure about, but I think `contains_key` avoid the deserialization of the data checking only if it exists

Reply:
> No completely sure about, but I think `contains_key` avoid the deserialization of the data checking only if it exists

+1 from my side. If we don't act on the returned value, we should always check with `contains_key` instead of `get`. This API could be called quite frequently, so any simple optimization is very valuable.

Reply:
Good call, thanks!